services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/taydennys/TaydennysAttachmentTransferService.kt

```diff
@@ -0,0 +1,39 @@
+package fi.hel.haitaton.hanke.taydennys
+
+import fi.hel.haitaton.hanke.attachment.common.ApplicationAttachmentEntity
+import fi.hel.haitaton.hanke.attachment.common.ApplicationAttachmentRepository
+import fi.hel.haitaton.hanke.attachment.common.TaydennysAttachmentMetadata
+import fi.hel.haitaton.hanke.attachment.common.TaydennysAttachmentRepository
+import fi.hel.haitaton.hanke.hakemus.HakemusEntity
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Propagation
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+class TaydennysAttachmentTransferService(
+    private val taydennysAttachmentRepository: TaydennysAttachmentRepository,
+    private val hakemusAttachmentRepository: ApplicationAttachmentRepository,
+) {
+
+    /** Transfers täydennys attachment to hakemus. */
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    fun transferAttachmentToHakemus(
```

Comment:
Couldn't this be in the metadata service?

Reply:
Yes, it could and it would be better place for it. Moved it there.